op-bindings/predeploys/addresses.go

```diff
@@ -3,66 +3,72 @@ package predeploys
 
 const (
-	L2ToL1MessagePasser           = "0x4200000000000000000000000000000000000016"
-	DeployerWhitelist             = "0x4200000000000000000000000000000000000002"
-	LegacyERC20ETH                = "0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000"
-	WETH9                         = "0x4200000000000000000000000000000000000006"
-	L2CrossDomainMessenger        = "0x4200000000000000000000000000000000000007"
-	L2StandardBridge              = "0x4200000000000000000000000000000000000010"
-	SequencerFeeVault             = "0x4200000000000000000000000000000000000011"
-	OptimismMintableERC20Factory  = "0x4200000000000000000000000000000000000012"
-	L1BlockNumber                 = "0x4200000000000000000000000000000000000013"
-	GasPriceOracle                = "0x420000000000000000000000000000000000000F"
-	L1Block                       = "0x4200000000000000000000000000000000000015"
-	GovernanceToken               = "0x4200000000000000000000000000000000000042"
-	LegacyMessagePasser           = "0x4200000000000000000000000000000000000000"
-	L2ERC721Bridge                = "0x4200000000000000000000000000000000000014"
-	OptimismMintableERC721Factory = "0x4200000000000000000000000000000000000017"
-	ProxyAdmin                    = "0x4200000000000000000000000000000000000018"
-	BaseFeeVault                  = "0x4200000000000000000000000000000000000019"
-	L1FeeVault                    = "0x420000000000000000000000000000000000001a"
+	L2ToL1MessagePasserAddr           = "0x4200000000000000000000000000000000000016"
+	DeployerWhitelistAddr             = "0x4200000000000000000000000000000000000002"
+	LegacyERC20ETHAddr                = "0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000"
+	WETH9Addr                         = "0x4200000000000000000000000000000000000006"
+	L2CrossDomainMessengerAddr        = "0x4200000000000000000000000000000000000007"
+	L2StandardBridgeAddr              = "0x4200000000000000000000000000000000000010"
+	SequencerFeeVaultAddr             = "0x4200000000000000000000000000000000000011"
+	OptimismMintableERC20FactoryAddr  = "0x4200000000000000000000000000000000000012"
+	L1BlockNumberAddr                 = "0x4200000000000000000000000000000000000013"
+	GasPriceOracleAddr                = "0x420000000000000000000000000000000000000F"
+	L1BlockAddr                       = "0x4200000000000000000000000000000000000015"
+	GovernanceTokenAddr               = "0x4200000000000000000000000000000000000042"
+	LegacyMessagePasserAddr           = "0x4200000000000000000000000000000000000000"
+	L2ERC721BridgeAddr                = "0x4200000000000000000000000000000000000014"
+	OptimismMintableERC721FactoryAddr = "0x4200000000000000000000000000000000000017"
+	ProxyAdminAddr                    = "0x4200000000000000000000000000000000000018"
+	BaseFeeVaultAddr                  = "0x4200000000000000000000000000000000000019"
+	L1FeeVaultAddr                    = "0x420000000000000000000000000000000000001a"
 )
 
+type Predeploy struct {
```

Comment:
What is the purpose of this struct?

Reply:
Ah ok I see its used in building the genesis. I'd like to delete the migration code from the codebase to prevent such large diffs

Reply:
Right, this PR has two commits, the first is the extension of predeploy type with these extra metadata fields, and the second is the crux of the change that implements the new non-migration L2 genesis w/ optional governance (which has a much smaller diff).  Consider the first commit tentative -- if predeploy metadata isn't likely to be useful beyond genesis we can revert & keep that metadata local to package genesis.

Reply:
Should I go ahead and delete the migration code? That would indeed simplify a lot of things....